frontend/src/scenes/notebooks/Nodes/NodeWrapper.tsx

```diff
@@ -85,27 +78,22 @@ function NodeWrapper<T extends CustomNotebookNodeAttributes>(
     useWhyDidIRender('NodeWrapper.props', props)
 
     const mountedNotebookLogic = useMountedLogic(notebookLogic)
-    const { isEditable, editingNodeId } = useValues(notebookLogic)
-    const { setTextSelection } = useActions(notebookLogic)
+    const { isEditable, editingNodeId } = useValues(mountedNotebookLogic)
 
-    // nodeId can start null, but should then immediately be generated
-    const nodeId = attributes.nodeId
-    const nodeLogicProps: NotebookNodeLogicProps = {
-        node,
-        nodeType,
-        attributes,
-        updateAttributes,
-        nodeId,
+    const logicProps: NotebookNodeLogicProps = {
+        ...props,
         notebookLogic: mountedNotebookLogic,
-        getPos,
-        resizeable: resizeableOrGenerator,
-        settings,
-        startExpanded,
-        titlePlaceholder,
     }
-    const nodeLogic = useMountedLogic(notebookNodeLogic(nodeLogicProps))
+
+    // nodeId can start null, but should then immediately be generated
+    const nodeLogic = useMountedLogic(notebookNodeLogic(logicProps))
     const { resizeable, expanded, actions } = useValues(nodeLogic)
-    const { setExpanded, deleteNode, toggleEditing } = useActions(nodeLogic)
+    const { setExpanded, deleteNode, toggleEditing, insertOrSelectNextLine, unmount } = useActions(nodeLogic)
+
+    useEffect(() => {
+        // TRICKY: child nodes mount the parent logic so we need to control the mounting / unmounting directly in this component
```

Comment:
It's not so much that they mount it but more that `useValues` or whatever kea thing will keep it mounted. Because we render the components outside of this heirachy, when the node is removed it doesn't remove the children from React so there are still references to the logic. 

If we instead forcefully remove it from the list of logics when the actual node unmounts then React re-renders the children, eventually unmounting the nodeLogic when there are no other references left.